core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -140,6 +139,7 @@ public void initialize(String name, Map<String, String> properties) {
       if (initializeCatalogTables) {
         initializeCatalogTables();
       }
+      checkAndMaybeUpdateSchema();
```

Comment:
I think the error handling is incorrect here. The errors state that the schema could not be initialized, but that applies only to `initializeCatalogTables`. I think the try/catch should move into the `initializeCatalogTables` method and `updateSchemaIfRequired` should have its own error handling.

Reply:
Correct, thanks I will fix that.